pkg/api/types.go

```diff
@@ -58,6 +64,10 @@ type Options struct {
 	PinPatch *int64 `json:"pin-patch,omitempty"`
 
 	RegexMatcher *regexp.Regexp `json:"-"`
+
+	// Architecture and OS to search for
+	Architecture *string `json:"pin-architecture,omitempty"`
+	OS           *string `json:"pin-os,omitempty"`
```

Comment:
Let's type these strings;

```
type Architecture *string
type OS *string
```

Reply:
Do we want to change it in the ImageTag also?